packages/schema-type/src/typescript/arrayAbiTypes.ts

```diff
@@ -1,17 +1,42 @@
 import { StaticAbiType } from "./schemaAbiTypes";
+import { isStaticAbiType } from "./staticAbiTypes";
 
-const arrayAbiTypePattern = /\[\]$/;
+export type ArrayAbiType = `${StaticAbiType}[]`;
+export type FixedArrayAbiType = `${StaticAbiType}[${number}]`;
+
+const arrayPattern = /\[\]$/;
+const fixedArrayPattern = /\[\d+\]$/;
 
 export type ArrayAbiTypeToStaticAbiType<T extends string> = T extends `${infer StaticAbiType}[]`
   ? StaticAbiType
   : never;
 
-export function isArrayAbiType<T extends `${StaticAbiType}[]`>(abiType: string): abiType is T {
-  return arrayAbiTypePattern.test(abiType);
+export type FixedArrayAbiTypeToStaticAbiType<T extends string> = T extends `${infer StaticAbiType}[${number}]`
+  ? StaticAbiType
+  : never;
+
+export function isArrayAbiType<T extends ArrayAbiType>(abiType: string): abiType is T {
+  return arrayPattern.test(abiType) && isStaticAbiType(abiType.replace(arrayPattern, ""));
+}
+
+export function isFixedArrayAbiType<T extends FixedArrayAbiType>(abiType: string): abiType is T {
```

Comment:
same here, would expect something like this:
```suggestion
export function isFixedArrayAbiType<T>(abiType: T): abiType is T & FixedArrayAbiType {
```